debian/snapd.prerm

```diff
@@ -0,0 +1,16 @@
+#! /bin/sh
+# prerm script for snapd
+
+set -e
+
+if [ "$1" = "remove" -o "$1" = "upgrade" ]; then
```

Comment:
Fixed according to your first comment, and resorted to stop and disable. --now is not available as pointed out in an earlier comment.